apps/src/code-studio/sectionsRedux.js

```diff
@@ -0,0 +1,49 @@
+import Immutable from 'immutable';
+import _ from 'lodash';
+
+// Action types
+export const SET_SECTIONS = 'sections/SET_SECTIONS';
+export const SELECT_SECTION = 'sections/SELECT_SECTION';
+
+// Action Creators
+export const setSections = sections => ({
+  type: SET_SECTIONS,
+  sections
+});
+
+export const selectSection = sectionId => ({
+  type: SELECT_SECTION,
+  sectionId
+});
+
+const SectionData = Immutable.Record({
+  selectedSection: null,
+  sectionsLoaded: false,
+  sectionIds: [],
+  bySection: {}
```

Comment:
sectionIds is arguably redundant in that we could get the same by looking at Object.keys(bySection). Thoughts?

Also noteworthy that although I have a `bySection` in both this subtree and in stageLock, I only store `sectionIds` here, which felt right to me. Also not sure if there's a better name than `bySection`. Here in particular we're only storing a name for each section so instead of:
```
bySection: {
  "1":  {
      name: "section 1"
  }
}
```
This could be 
```
names: {
  "1":  "section 1"
}
```

I could see us storing more here at some point in the future, but maybe it makes more sense to just take the "names" approach until we need to store more?